pyflagser/flagser.py

```diff
@@ -86,18 +92,30 @@ def flagser(flag_matrix, min_dimension=0, max_dimension=np.inf, directed=True,
     if not approximation:
         approximation = -1
 
-    edges = sp.coo_matrix(flag_matrix, copy=True)
-    edges.setdiag(np.nan)
+    if type(flag_matrix) is np.ndarray:
+        np.fill_diagonal(flag_matrix, np.nan)
```

Comment:
I'm a little worried by this line. It seems that in this way the diagonal values of `flag_matrix` are forgotten. But the user may have inputted diagonal values deliberately to signify when nodes enter the filtration. Notice that e.g. `ripser` is able to deal with non-zero diagonal values no problem (they can be non-zero and even negative), and computes persistent homology correctly in that case. @gtauzin am I missing something?

Reply:
This is has changed in #30 because this was not working for non-float dtype. But the diagonal elements are stored in vertices beforehand.